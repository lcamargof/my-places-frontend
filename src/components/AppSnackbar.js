import React from 'react';
import PropTypes from 'prop-types';
import { IconButton, Snackbar, withStyles } from 'material-ui';
import CloseIcon from '@material-ui/icons/Close';

const styles = theme => ({
  close: {
    width: theme.spacing.unit * 4,
    height: theme.spacing.unit * 4,
  },
});

export const AppSnackbar = ({ classes, text, onClose, open }) => (
  <Snackbar
    anchorOrigin={{
      vertical: 'bottom',
      horizontal: 'left',
    }}
    open={open}
    autoHideDuration={6000}
    onClose={onClose}
    SnackbarContentProps={{
      'aria-describedby': 'snackbar-msg',
    }}
    message={<span id="snackbar-msg">{ text }</span>}
    action={[
      <IconButton
        key="close"
        aria-label="Close"
        id="close-snackbar"
        color="inherit"
        className={classes.close}
        onClick={onClose}
      >
        <CloseIcon />
      </IconButton>,
    ]}
  />
);

AppSnackbar.propTypes = {
  classes: PropTypes.object.isRequired,
  text: PropTypes.string.isRequired,
  onClose: PropTypes.func.isRequired,
  open: PropTypes.bool.isRequired,
};

export default withStyles(styles)(AppSnackbar);