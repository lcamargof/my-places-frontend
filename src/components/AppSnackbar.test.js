import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { AppSnackbar } from './AppSnackbar';

describe('Component: AppSnackbar', () => {
  it('Should render correctly', () => {
    const component = shallow(<AppSnackbar classes={{}} text="test" onClose={jest.fn()} open={true} />);
    expect(shallowToJson(component)).toMatchSnapshot();
  });
});