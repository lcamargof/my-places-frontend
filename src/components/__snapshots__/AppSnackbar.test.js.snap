// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: AppSnackbar Should render correctly 1`] = `
<WithStyles(Snackbar)
  SnackbarContentProps={
    Object {
      "aria-describedby": "snackbar-msg",
    }
  }
  action={
    Array [
      <WithStyles(IconButton)
        aria-label="Close"
        className={undefined}
        color="inherit"
        id="close-snackbar"
        onClick={[Function]}
    >
        <pure(Close) />
    </WithStyles(IconButton)>,
    ]
  }
  anchorOrigin={
    Object {
      "horizontal": "left",
      "vertical": "bottom",
    }
  }
  autoHideDuration={6000}
  message={
    <span
      id="snackbar-msg"
    >
      test
    </span>
  }
  onClose={[Function]}
  open={true}
/>
`;
